other/scoring_algorithm.py

```diff
@@ -0,0 +1,106 @@
+'''
+developed by: markmelnic
+original repo: https://github.com/markmelnic/Scoring-Algorithm
+
+Analyse data using a range based procentual proximity algorithm
+and calculate the linear maximum likelihood estimation.
+The basic principle is that all values supplied will be broken
+down to a range from 0 to 1 and each column's score will be added
+up to get the total score.
+'''
+
+def scalg(source_data, weights):
+    '''
+    weights - int list
+    possible values - 0 / 1
+    0 if lower values have higher weight in the data set
+    1 if higher values have higher weight in the data set
+    ==========
+    Example for data of vehicles
+    price|mileage|registration_year
+    20k  |60k    |2012
+    22k  |50k    |2011
+    23k  |90k    |2015
+    16k  |210k   |2010
+    
+    We want the vehicle with the lowest price,
+    lowest mileage but newest registration year.
+    Thus the weights for each column are as follows:
+    [0, 0, 1]
+    
+    >>> score([[20k, 60k, 2012], [22k, ...etc] ...etc], [0, 0, 1])
+    
+    [[20000, 60000, 2012, 1.7660714285714287], --> highest score
+    [22000, 50000, 2011, 1.3428571428571427],
+    [23000, 90000, 2015, 1.75],
+    [16000, 210000, 2010, 1.0]] --> lowest score
+    '''
+    
+    # getting data
+    data_lists = []
+    for item in source_data:
+        for i in range(len(item)):
+            try:
+                data_lists[i].append(float(item[i]))
+            except:
```

Comment:
Bare exceptions are not accepted in this repo.
See PEP8 and https://realpython.com/the-most-diabolical-python-antipattern/